### PR TITLE
Add config support in codegen

### LINKED ISSUE #13: Speed up unit tests
**User Story**

Most of the time of the unit test jobs is spent on installing the tools and generating the code that is automatically generated. We should bake in all tools and not regenerate the code as it should be included in the repository.

/kind feature

### PR DESCRIPTION
**Pull request by digambar15**
Add all config files in codegen scripts.

Fixes #13

### CODE REVIEW

File: hack/codegen.sh [hunk 1]
```diff
@@ -10,7 +10,7 @@ if [ "${IS_CONTAINER}" != "false" ]; then
   export XDG_CACHE_HOME=/tmp/.cache
   eval "$(go env)"
   cd "${GOPATH}"/src/github.com/metal3-io/cluster-api-provider-metal3
-  INPUT_FILES="api/v1alpha2/zz_generated.*.go api/v1alpha3/zz_generated.*.go api/v1alpha4/zz_generated.*.go baremetal/mocks/zz_generated.*.go"
+  INPUT_FILES="config/certmanager/*.yaml config/crd/*.yaml config/crd/bases/*.yaml config/crd/patches/*.yaml config/default/*.yaml config/ipam/*.yaml config/manager/*.yaml config/rbac/*.yaml config/webhook/*.yaml config/*.yaml api/v1alpha2/zz_generated.*.go api/v1alpha3/zz_generated.*.go api/v1alpha4/zz_generated.*.go baremetal/mocks/zz_generated.*.go"
```
Comment (maelk) [inline hack/codegen.sh:13]: can you break this ?
```suggestion
  INPUT_FILES="\
  config/certmanager/*.yaml \
  config/crd/*.yaml \
  config/crd/bases/*.yaml \
  config/crd/patches/*.yaml \
  config/default/*.yaml \
  config/ipam/*.yaml \
  config/manager/*.yaml \
  config/rbac/*.yaml \
  config/webhook/*.yaml \
  config/*.yaml \
  api/v1alpha2/zz_generated.*.go \
  api/v1alpha3/zz_generated.*.go \
  api/v1alpha4/zz_generated.*.go \
  baremetal/mocks/zz_generated.*.go"
```

Comment (digambar15) [reply hack/codegen.sh:13]: okay

Comment (digambar15) [reply hack/codegen.sh]: Done!

**Pull request merged**